shell/platform/windows/win32_flutter_window.h

```diff
@@ -0,0 +1,177 @@
+  // down to up
+  void OnPointerUp(double x, double y) override;
+
+  // Virtual method called by base window when character input occurs.
+  void OnChar(unsigned int code_point) override;
+
+  // Virtual method called by base window when raw keyboard input occurs.
+  void OnKey(int key, int scancode, int action, int mods) override;
+
+  // Virtual method called by base window when mouse scrollwheel input occurs.
+  void OnScroll(double delta_x, double delta_y) override;
+
+  // Virtual methed called by the base window when the user closes the Windows
+  void OnClose();
+
+  // Configures the window instance with an instance of a running Flutter engine
+  // returning a configured FlutterDesktopWindowControllerRef.
+  FlutterDesktopWindowControllerRef SetState(FLUTTER_API_SYMBOL(FlutterEngine)
+                                                 state);
+
+  // Returns the currently configured Plugin Registrar.
+  FlutterDesktopPluginRegistrarRef GetRegistrar();
+
+  // Callback passed to Flutter engine for notifying window of platform
+  // messages.
+  void HandlePlatformMessage(const FlutterPlatformMessage*);
+
+  // Create a surface for Flutter engine to render into.
+  void CreateRenderSurface();
+
+  // Destroy current rendering surface if one has been allocated.
+  void DestroyRenderSurface();
+
+  // Callbacks for clearing context, settings context and swapping buffers.
+  bool ClearContext();
+  bool MakeCurrent();
+  bool SwapBuffers();
+
+  // Sends a window metrics update to the Flutter engine using current window
+  // dimensions in physical
+  void SendWindowMetrics();
+
+ private:
+  // Reports a mouse movement to Flutter engine.
+  void SendPointerMove(double x, double y);
+
+  // Reports mouse press to Flutter engine.
+  void SendPointerDown(double x, double y);
+
+  // Reports mouse release to Flutter engine.
+  void SendPointerUp(double x, double y);
+
+  // Reports a keyboard character to Flutter engine.
+  void SendChar(unsigned int code_point);
+
+  // Reports a raw keyboard message to Flutter engine.
+  void SendKey(int key, int scancode, int action, int mods);
+
+  // Reports scroll wheel events to Flutter engine.
+  void SendScroll(double delta_x, double delta_y);
+
+  // Updates |event_data| with the current location of the mouse cursor.
+  void SetEventLocationFromCursorPosition(FlutterPointerEvent* event_data);
+
+  // Set's |event_data|'s phase to either kMove or kHover depending on the
+  // current
+  // primary mouse button state.
+  void SetEventPhaseFromCursorButtonState(FlutterPointerEvent* event_data);
+
+  // Sends a pointer event to the Flutter engine based on givern data.  Since
+  // all input messages are passed in physical pixel values, no translation is
+  // needed before passing on to engine.
+  void SendPointerEventWithData(const FlutterPointerEvent& event_data);
+
+  std::unique_ptr<AngleSurfaceManager> surface_manager{nullptr};
```

Comment:
Same question as in the other file about why {} vs =

Reply:
Fixed in next commit